### DISCUSSION TITLE
Component update grouping with exceptions & how to turn an exception into a valid 200 response.

### DISCUSSION THREAD

**Discussion by booni3** [livewire/livewire#3100] [7. Bugs (archived)]
### Description

I assume by design, Livewire seems to group updates that happen to a component in quick succession (made more frequent by a slow internet).

In our case, we have a handheld barcode scanner that is sending a request on every scan.

![image](https://user-images.githubusercontent.com/20134485/100460027-661c9200-30be-11eb-86d2-9e47ea4d5c23.png)

In this example, we have a method that adds a product to a list and then throws an exception if we scan too many.

```php
if ($this->isOverPacked($product_uuid)) {
    throw NotifyScanUser::overPacked();
}
```

The issue I am facing here is that when we throw an exception and we have grouped requests, every request fails. Actually what should happen is that the first 4 get 200 responses and update the component and then the 5th, 6th, 7th etc. all fail.

I realise this is probably not a livewire bug and maybe its too edge case to include in the docs?

For my fix, I am currently doing:

```php
if ($this->isOverPacked($product_uuid]) {
    return $this->emit('notify', 'Item Over Packed');
}
```

This notifies the user that they have overpacked while not causing an exception. However, it needs to be used carefully as a return is obviously not going to stop the execution in the same way that throwing an exception will.

### What I want to achieve

I want to be able to stop the execution of the code at this point but return a 200 error with a message. I have tried intercepting it in the Laravel exception handler but it always throws some kind of redirect error within the console.

**Comment by robertvansteen** [comment]
Yeah would be really nice to just be able to throw an exception and then have some method on the component act as an error handler that allows you to handle the exception, by for example emitting an event to show some error message to the user.
